### PR TITLE
COVID-19-Data-Tracker

### PR DESCRIPTION
**Pull request by dami1080**
In this project, I:

- Created the React App.
- Selected the MMediaGroup Covid-19 API for Data
- Installed basic dependencies (React-Redux, Redux Logger and React Router).
- Created routes and navigations for the app
- Created the directory for all Redux state slice files.
- Created routes and view components for the Home and Details page.
- Fetched data from the API when the application starts.
- Dispatched an action to store the selected data in the Redux store.
- Used useSelector(), Redux Hook, to select the state slices and render countries and country details in corresponding routes.
- Implemented the Filter for filtering the number of confirmed cases.
- Write unit tests with Jest and React Testing Library
- Styled the application using plain CSS and React Bootstrap.
- Integrated linters.

[video demo](https://www.loom.com/share/083e86989baa424a96f8ce70817db5c3)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <Router basename={process.env.PUBLIC_URL}>
+        <Switch>
+          <Route exact path="/">
+            <Home />
+          </Route>
+          <Route path="/details/:name">
+            <Details />
+          </Route>
+        </Switch>
+      </Router>
```
Comment (Davidosky007) [inline src/App.js]: * Kindly ensure The project is a single page application (SPA) using react-router-dom to serve each page without reloading as currently, it reloads when i visit the details page.

File: src/tests/TotalCard.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import renderer from 'react-test-renderer';
+import TotalCard from '../components/TotalCard';
+
+describe('Filter component', () => {
+  test('Snapshot test', () => {
+    const totalCard = renderer.create(
+      <TotalCard name="AFRICA" total="200987" />,
+    )
```
Comment (Davidosky007) [inline src/tests/TotalCard.test.js:8]: * Kindly perform a mocking data fetching and Redux Store connection as stated in the project [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=You%20may%20need,the%20Redux%20Store.).

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @dami1080 👋, 

Good job so far! 💪 
* No linter errors ✔️ 
* Descriptive commit messages ✔️ 
* Well written PR ✔️ 
* Proper use of git-flow ✔️ 
* Cool design ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
* Kindly add a highlight of some interesting piece of code or something you built that you are very proud of 🙏 .
_Check the comments under the review._
almost-jumpin-gif


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
wever, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/countries/countries.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      return countries;
+  }
+};
+
+export const filterCountry = (value) => async (dispatch) => {
+  const continent = await getContinent();
+  const countries = Object.values(continent);
+  const filteredCountry = filterNumbers(countries, value);
+  console.log(filteredCountry);
+
```
Comment (ArthurGC) [inline src/redux/countries/countries.js]: - [OPTIONAL] It would be better if you remove `console.log` to follow the best practices in coding. ♻️

**Review by ArthurGC — CHANGES_REQUESTED** [review]
# Required Changes ♻️ :warning: 
Hi  @dami1080 

Great work on making the changes requested by a previous reviewer. 👏 
You've done well implementing some of the requested changes but there are some issues that you still need to work on to go to the next project but you are almost there! 



## Good Points :+1: :
Great job following most of the first reviewer's suggestions :heavy_check_mark:

## Aspects to improve 🔴 
_Check the comments under the review._

- [ ] **As the previous reviewer said. Please, add a highlight of some interesting piece of code or something you built that you are very proud of in your video demo.**


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @dami1080, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**